### PR TITLE
Replace TM service reference with TX Mgr factory in JPAComponentImpl

### PR DESCRIPTION
**Pull request by dazavala**
For issue #25216 

The `JPAComponent` service component does not tolerate dynamic updates to the `TransactionManagerService`. Doing so causes `JPAComponentImpl` to reactivate, which is problematic whenever applications do not also restart. Alleviate the problem by modifying `JPAComponentImpl` to use the `EmbeddableTransactionManagerFactory (ETMF)`, instead of the `TransactionManagerService`, to access the `EmbeddableWebSphereTransactionManager` singleton and the `UOWCurrent` implementations. Notice the `TransactionManagerService` impl wraps the `ETMF` and exports the same behaviors.

This solution is preferable to modifying the `TMS` reference from `static+mandatory` to `dynamic+optional`, because the `TMS` is required for the Persistence/JPA and Persistence/JPA Container feature runtimes. However, doing so would solve the problem as declarative services would simply inject the updated service reference into the existing JPAComponentImpl rather than reactivate it (i.e. deactivate > unbind svc refs > rebind svc refs > reactivate.) Also, note that a `dynamic+mandatory` reference does not solve the problem.

The ideal solution would modify the `JPAComponentImpl` to tolerate service updates w/o requiring applications to restart, which IMHO, would require managing application state within some other class. Also, the solution must ensure the service component activates before it is used. This behavior can be enabled by 1) configuring the `JCAComponent` service component to `immediate`; 2) by tracking the `TMS` and consuming the `JPAComponent` service (i.e. getService(JPAC)) when the updated `TMS` service registers with OSGi; or 3) by changing the `JPAComponent` service component's dependency graph to ensure some service always consumes the `JPAComponent` service after deactivation, before it used to process work.

### CODE REVIEW

**Comment by dazavala** [comment]
#build

#libby

**Comment by dazavala** [comment]
#build

**Comment by dazavala** [comment]
@tkburroughs  Review concerns: These changes introduce import `com.ibm.tx.jta.embeddable.EmbeddableTransactionManagerFactory` into JPAComponentImpl, which requires the server to resolve the new import package in bundle `com.ibm.ws.jpa.container.jar`. Package `com.ibm.tx.jta.embeddable` is exported by `com.ibm.ws.tx.embeddable.jar`. Bundles that export `...tx.jta...` packages, including `com.ibm.ws.tx.embeddable.jar`, are provisioned by the transaction-x.y features, and these bundle manifests declare the necessary package exports. Since Persistence (JPA) features include transaction-x.y, this solution does not introduce potential packing wiring failures when the server resolves installed feature bundles.

**Comment by dazavala** [comment]
#build

**Review by tkburroughs — APPROVED** [review]

**Comment by dazavala** [comment]
#libby

Liberty personal builds passed. The InstantOn personal build request remains pending, so I manually verified this branch in my workspace by running the FULL checkpoint_FAT. Look good. Ready to merge pending libby's checks AND devOps authorization.

**Comment by dazavala** [comment]
FYI, the InstantOn personal build completed with no breaks related to these changes.
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_nPoRADDXEe6cYpJ4GUjoAA&tab=com.ibm.team.build.web.ui.internal.editors.result.overview.OverviewPage

This PR is ready to merge pending devOps approval.

**Pull request merged**
